### PR TITLE
refactor(InfoCard): use text truncate; add titleHeadingLevel as prop

### PR DESCRIPTION
**Pull request by sooinc**
### Overview

- Adds titleHeadingLevel as prop
- Replaces deprecated Truncate component

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: EGG-2083
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.42-alpha.9a0ee4.0<br/>@codecademy/gamut-labs@28.6.5-alpha.9a0ee4.0<br/>@codecademy/styleguide@53.0.3-alpha.9a0ee4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61e87b2525f7ef1ec802daac--gamut-preview.netlify.app>https://61e87b2525f7ef1ec802daac--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61e87b2525f7ef1ec802daac>Deploy Logs</a></p>

File: packages/gamut-labs/src/InfoCard/index.tsx [hunk 1]
```diff
@@ -32,21 +38,27 @@ const TopText: React.FC<TextProps> = ({ text }) => (
   </Text>
 );
 
-const Title: React.FC<TextProps> = ({ text }) => (
-  <Text as="h3" mb={4} fontSize={20}>
-    <Truncate lines={2}>{text}</Truncate>
+const Title: React.FC<TitleProps> = ({ text, headingLevel }) => (
+  <Text as={headingLevel} mb={4} fontSize={20}>
```
Comment (saghdaey) [inline packages/gamut-labs/src/InfoCard/index.tsx:42]: q: do we want headingLevel to always be required for this component, or have a default (like the former `h3` value?)

File: packages/gamut-labs/src/InfoCard/index.tsx [hunk 2]
```diff
@@ -32,21 +38,27 @@ const TopText: React.FC<TextProps> = ({ text }) => (
...
+    <Text truncate="ellipsis" truncateLines={2}>
+      {text}
+    </Text>
   </Text>
 );
 
 const Subtitle: React.FC<TextProps> = ({ text }) => (
   <Text variant="p-small" textColor="gray-900">
-    <Truncate lines={2}>{text}</Truncate>
+    <Text truncate="ellipsis" truncateLines={2}>
```
Comment (saghdaey) [inline packages/gamut-labs/src/InfoCard/index.tsx:51]: love 2 see it!!!!

**Review by saghdaey — APPROVED** [review]

File: packages/gamut-labs/src/InfoCard/index.tsx (hunk 1 above)
Comment (sooinc) [reply packages/gamut-labs/src/InfoCard/index.tsx:42]: The `Title` component is only being used by the `InfoCard` component which passes a default value of `h3` if `titleHeadingLevel` is not provided so `Title` will always be expected to have a `headingLevel`.

Comment (saghdaey) [reply packages/gamut-labs/src/InfoCard/index.tsx:42]: got it! thx :)

**Review by dreamwasp — APPROVED** [review]
🔥

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.42-alpha.962f2f.0<br/>@codecademy/gamut-labs@28.6.5-alpha.962f2f.0<br/>@codecademy/gamut@43.1.1-alpha.962f2f.0<br/>@codecademy/monaco@0.5.31-alpha.962f2f.0<br/>@codecademy/styleguide@53.0.3-alpha.962f2f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61eaced88995c109bd6caf6a--gamut-preview.netlify.app>https://61eaced88995c109bd6caf6a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61eaced88995c109bd6caf6a>Deploy Logs</a></p>

**Pull request merged**